### PR TITLE
Refactor/better naming conventions/organised file structure/add unit test cases

### PR DESCRIPTION
**Pull request by hirdesharma**
Followed java naming conventions and file structure for improving readability and understandability of code
Unit Test cases
1. Added tests for ArgumentParser to ensure correct parsing and handling of command-line arguments.
2. Created tests for Item class to validate item creation and tax calculation.
3. Developed tests for ItemValidator to check validation of input arguments.
4. Included tests for ItemType enum to ensure correct type conversion and handling.

### CODE REVIEW

File: src/main/java/assignment1/utils/ArgumentParser.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package main.java.assignment1.utils;
+
+import main.java.assignment1.exceptions.InvalidArgument;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ArgumentParser {
```
Comment (sankethkini) [inline src/main/java/assignment1/utils/ArgumentParser.java:8]: change it to end with Utils

File: src/test/java/ArgumentParserTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import main.java.assignment1.exceptions.InvalidArgument;
+import main.java.assignment1.utils.ArgumentParser;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ArgumentParserTest {
```
Comment (sankethkini) [inline src/test/java/ArgumentParserTest.java:10]: properly place the testcase

File: src/main/java/assignment1/models/TaxCalculator.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package main.java.assignment1.models;
+
+import main.java.assignment1.enums.ItemType;
+
+public class TaxCalculator {
```
Comment (sankethkini) [inline src/main/java/assignment1/models/TaxCalculator.java:5]: this service not model

Comment (hirdesharma) [reply src/main/java/assignment1/models/TaxCalculator.java:5]: created a new package called services and moved TaxCalculator there and renamed it to TaxCalcutorService

File: src/main/java/assignment1/utils/ArgumentParser.java (hunk 1 above)
Comment (hirdesharma) [reply src/main/java/assignment1/utils/ArgumentParser.java:8]: renamed as ArgumentParserUtils

File: src/test/java/ArgumentParserTest.java (hunk 2 above)
Comment (hirdesharma) [reply src/test/java/ArgumentParserTest.java:10]: placed the testcases in proper file structure same as their class folder name

**Pull request merged**